### PR TITLE
softmax cross entropy support high rank

### PR DESCRIPTION
**Pull request by phlrain**
softmax cross entopy support high rank

### CODE REVIEW

File: paddle/fluid/operators/softmax_with_cross_entropy_op.cc [hunk 1]
```diff
@@ -106,24 +106,40 @@ class SoftmaxWithCrossEntropyOp : public framework::OperatorWithKernel {
...
-        "The input of softmax_with_cross_entropy should be a 2-D tensor.");
-    PADDLE_ENFORCE_EQ(labels_dims.size(), 2UL,
-                      "The labels should be a 2-D tensor.");
+        rank, labels_dims.size(),
+        "Input(logits) and Input(Label) shall have the same rank.");
+    bool check = true;
+    if ((!ctx->IsRuntime()) && (framework::product(logits_dims) <= 0 ||
+                                framework::product(labels_dims) <= 0)) {
+      check = false;
+    }
```
Comment (chengduoZH) [inline paddle/fluid/operators/softmax_with_cross_entropy_op.cc]:
```
bool check = ctx->IsRuntime() || (framework::product(logits_dims) > 0 &&  framework::product(labels_dims) > 0);
```

File: paddle/fluid/operators/softmax_with_cross_entropy_op.cc [hunk 2]
```diff
@@ -106,24 +106,40 @@ class SoftmaxWithCrossEntropyOp : public framework::OperatorWithKernel {
...
+                        "If Attr(softLabel) == false, the last dimension of "
                         "Input(Label) should be 1.");
     }
 
     ctx->SetOutputDim("Softmax", logits_dims);
-    ctx->SetOutputDim("Loss", {logits_dims[0], 1});
+    auto loss_dims = logits_dims;
+    loss_dims[rank - 1] = 1;
+    ctx->SetOutputDim("Loss", loss_dims);
+    // ctx->SetOutputDim("Loss", {logits_dims[0], 1});
```
Comment (chengduoZH) [inline paddle/fluid/operators/softmax_with_cross_entropy_op.cc]: line 142 should be removed.

**Comment by kuke** [comment]
No more comments

**Review by kuke — APPROVED** [review]
